rhel/vex/parser.go

```diff
@@ -0,0 +1,454 @@
+	rc                 *repoCache
+}
+
+// KnownAffectedVulnerabilities processes the "known_affected" array of products
+// in the VEX object.
+func (c *creator) knownAffectedVulnerabilities(ctx context.Context, v csaf.Vulnerability, protoVulnFunc func() *claircore.Vulnerability) ([]*claircore.Vulnerability, error) {
+	out := []*claircore.Vulnerability{}
+	for _, pc := range v.ProductStatus["known_affected"] {
+		rel := c.c.ProductTree.Relationships.FindRelationship(pc, "default_component_of")
+		if rel == nil {
+			// Should never get here, error in data
+			zlog.Debug(ctx).
+				Str("product:package", pc).
+				Msg("could not find a relationship for product:package")
+			continue
+		}
+		if strings.HasPrefix(rel.ProductRef, "kernel") {
+			// We don't want to ingest kernel advisories as
+			// containers have no say in the kernel.
+			continue
+		}
+		repoProd := c.pc.Get(c.c, rel.RelatesToProductRef)
+		if repoProd == nil {
+			// Should never get here, error in data
+			zlog.Warn(ctx).
+				Str("prod", rel.RelatesToProductRef).
+				Msg("could not find product in product tree")
+			continue
+		}
+		cpeHelper, ok := repoProd.IdentificationHelper["cpe"]
+		if !ok {
+			zlog.Warn(ctx).
+				Str("prod", rel.RelatesToProductRef).
+				Msg("could not find cpe helper type in product")
+			continue
+		}
+
+		// pkgName will be overridden if we find a valid pURL
+		pkgName := rel.ProductRef
+		compProd := c.pc.Get(c.c, rel.ProductRef)
+		if compProd == nil {
+			// Should never get here, error in data
+			zlog.Warn(ctx).
+				Str("pkg", rel.ProductRef).
+				Msg("could not find package in product tree")
+			continue
+		}
+		// It is possible that we will not find a pURL, in that case
+		// the package.Name will be reported as-is.
+		purlHelper, ok := compProd.IdentificationHelper["purl"]
+		if ok {
+			purl, err := packageurl.FromString(purlHelper)
+			switch {
+			case err != nil:
+				zlog.Warn(ctx).
+					Str("purl", purlHelper).
+					Err(err).
+					Msg("could not parse PURL")
+			default:
+				pkgName = purl.Name
+			}
+			if purl.Type != packageurl.TypeRPM || purl.Namespace != "redhat" {
+				// Just ingest advisories that are Red Hat RPMs, this will
+				// probably change down the line when we consolidate updaters.
+				continue
+			}
+		}
+
+		vuln := protoVulnFunc()
+		// What is the deal here? Just stick the package name in and f-it?
+		// That's the plan so far as there's no PURL product ID helper.
+		vuln.Package = &claircore.Package{
+			Name: pkgName,
+			Kind: claircore.SOURCE,
+		}
+
+		wfn, err := cpe.Unbind(cpeHelper)
+		if err != nil {
+			return nil, fmt.Errorf("could not unbind cpe: %s %w", cpeHelper, err)
+		}
+		vuln.Repo = c.rc.Get(wfn)
+		if sc := c.c.FindScore(pc); sc != nil {
+			vuln.Severity, err = cvssVectorFromScore(sc)
+			if err != nil {
+				return nil, fmt.Errorf("could not parse CVSS score: %w, file: %s", err, c.vulnLink)
+			}
+		}
+		if t := c.c.FindThreat(pc, "impact"); t != nil {
```

Comment:
@RTann @hdonnay Here we are getting the threat and converting to a `claircore.Severity` as discussed. There are currently a few files that are known to have no `threat` object, we are waiting on RH prodsec for instructions on how to deal with those / if that is a bug. Right now the severity will just be `Unknown` and the logs will warn.